### PR TITLE
Introduce mqtt_default_vhosts global parameter

### LINKED ISSUE #73: When authenticating using a client cert, there is no way to specify a vhost for that client
When connecting with username/password, vhost is specified as part of username (vhost:username), but there is no username when authenticating with a client cert. Is there some way to tie a vhost to a user using client cert authentication?

Some possible solutions are:

1) Use the x509 certificate's optional fields. This could be problematic for many users.
2) Provide a mapping of user (via client cert) to vhost via runtime parameters.

### PR DESCRIPTION
**Pull request by acogoluegnes**
Fixes #73

### CODE REVIEW

File: test/auth_SUITE.erl [hunk 1]
```diff
@@ -72,12 +78,21 @@ mqtt_config(no_ssl_user) ->
...
-    User = "O=client,CN=" ++ Hostname,
-    {ok,_} = rabbit_ct_broker_helpers:rabbitmqctl(Config, 0, ["add_user", User, ""]),
-    {ok, _} = rabbit_ct_broker_helpers:rabbitmqctl(Config, 0, ["set_permissions",  "-p", "/", User, ".*", ".*", ".*"]),
-    Config1 = rabbit_ct_helpers:set_config(Config, [{temp_ssl_user, User}]),
+    Config1 = set_cert_user_on_default_vhost(Config),
     rabbit_ct_helpers:testcase_started(Config1, Testcase);
+init_per_testcase(ssl_user_vhost_success, Config) ->
+    Config1 = set_cert_user_on_default_vhost(Config),
+    User = ?config(temp_ssl_user, Config1),
+    {ok, _} = rabbit_ct_broker_helpers:rabbitmqctl(Config1, 0, ["clear_permissions",  "-p", "/", User]),
```
Comment (michaelklishin) [inline test/auth_SUITE.erl]: I'd introduce a new helper [like this one](https://github.com/rabbitmq/rabbitmq-ct-helpers/blob/master/src/rabbit_ct_broker_helpers.erl#L1070).

File: test/auth_SUITE.erl [hunk 2]
```diff
@@ -86,21 +101,65 @@ init_per_testcase(user_credentials_auth, Config) ->
...
+
+set_vhost_for_cert_user(Config, User) ->
+    VhostForCertUser = <<"vhost_for_cert_user">>,
+    UserToVHostMappingParameter = [
+        {rabbit_data_coercion:to_binary(User), VhostForCertUser},
+        {<<"O=client,CN=unlikelytoexistuser">>, <<"vhost2">>}
+    ],
+    ok = rabbit_ct_broker_helpers:add_vhost(Config, VhostForCertUser),
+    ok = rabbit_ct_broker_helpers:set_full_permissions(Config, rabbit_data_coercion:to_binary(User), VhostForCertUser),
+    ok = rabbit_ct_broker_helpers:rpc(
```
Comment (michaelklishin) [inline test/auth_SUITE.erl]: Also worth adding a CT helper.

File: test/auth_SUITE.erl [hunk 3]
```diff
@@ -86,21 +101,65 @@ init_per_testcase(user_credentials_auth, Config) ->
...
 end_per_testcase(Testcase, Config) when Testcase == ssl_user_auth_success;
-                                        Testcase == ssl_user_auth_failure ->
-    User = ?config(temp_ssl_user, Config),
-    {ok,_} = rabbit_ct_broker_helpers:rabbitmqctl(Config, 0, ["delete_user", User]),
+                                        Testcase == ssl_user_auth_failure;
+                                        Testcase == ssl_user_vhost_not_allowed ->
+    delete_cert_user(Config),
     rabbit_ct_helpers:testcase_finished(Config, Testcase);
+end_per_testcase(TestCase, Config) when TestCase == ssl_user_vhost_success;
+                                        TestCase == ssl_user_vhost_failure->
```
Comment (michaelklishin) [inline test/auth_SUITE.erl]: The missing space before the `->` here is fairly confusing.

**Comment by acogoluegnes** [comment]
* Happy path => `ssl_user_vhost_parameter_mapping_success`
 * Default vhost is not configured via global params => `ssl_user_auth_success`
 * Default vhost is configured via GP but does not exist => `ssl_user_vhost_parameter_mapping_vhost_does_not_exist`
 * Default vhost is configured via GP but the user has no permissions => `ssl_user_vhost_parameter_mapping_not_allowed`
 * Default vhost is not configured via GP, is configured via config file but the user has no permissions => `ssl_user_vhost_not_allowed`

**Pull request merged**